File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
import sys
input = sys.stdin.readline
n = int(input())
a = list(map(int, input().split()))
x = int(input())
a.sort()
start, end = 0, n-1
cnt = 0
while(start < end):
    if a[start] + a[end] > x: end -= 1
    elif a[start] + a[end] < x: start += 1
    elif a[start] + a[end] == x:
        cnt += 1
        start += 1
        end -= 1
print(cnt)

File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int n, x, cnt=0;
    cin >> n;
    int a[n];
    for (int i = 0; i < n; i++) cin >> a[i];
    cin >> x;

    sort(a, a+n);
    int i=0, j=n-1;
    while (i < j) {
        int t = a[i] + a[j];
        if (t > x) j--;
        else if (t < x) i++;
        else {cnt++; i++; j--;}
    }
    cout << cnt;
    return 0;
}